### PR TITLE
MES-1531 - Configuration Microservice with Dummy Data

### PR DESCRIPTION
**Pull request by Cavalier07**
Initial Version of Configuration Microservice.

Will return a dummy config object to any requests to /configuration/{scope} with scope being dev, test or prod.

### CODE REVIEW

**Comment by ammarhaiderbjss** [comment]
Looks good, are we going to return _a_ bunch of json so we can verify we can do that

File: src/functions/get/domain/config.mock.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export const config = {
+  GoogleAnalyticsID: 'dev-ga-id',
```
Comment (johnmorleybjss) [inline src/functions/get/domain/config.mock.ts]: What's the reason for InitCaps in here vs camelCase in the app?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,46 @@
...
+All serverless functions live in dedicated directories in `src/functions`.
+Code that is common between multiple functions should reside in `src/common`.
+
+As per the principles of Hexagonal Architecture, each function has the following directories to help us separate concerns:
+
+* `framework` - contains all Inbound and Outbound Adapters, and all use of external/proprietary APIs - depends upon...
+* `application` - contains all Inbound and Outbound Ports, doesn't use any external/proprietary APIs - depends upon...
+* `domain` - contains all domain objects (Aggregates, Objects, Value classes etc) with all "business logic" (not just anaemic data holders), doesn't use any external/proprietary APIs.
+
+## Bootstrap
```
[the remark this answers is not in the document]
Comment (Cavalier07) [reply README.md]: Removed

File: src/common/domain/README.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Dynamically generated schema interface goes here
```
[the remark this answers is not in the document]
Comment (Cavalier07) [reply src/common/domain/README.md]: Removed

File: src/functions/get/domain/config.mock.ts (hunk 1 above)
Comment (Cavalier07) [reply src/functions/get/domain/config.mock.ts]: Have changed to camelCase, thanks for spotting :)

File: src/functions/get/domain/config.mock.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+export const config = {
+  googleAnalyticsID: 'dev-ga-id',
+  journal : {
+    numberOfDaysToShow: 2,
```
Comment (szabolcs-keresztes) [inline src/functions/get/domain/config.mock.ts]: This has to be at least 3 for being able to navigate through the weekend

Comment (buzzub) [reply src/functions/get/domain/config.mock.ts]: That is really the number of working days. Saturday and Sunday are treated in TARS as non working days. See the examples at the bottom of https://wiki.i-env.net/display/MES/Journal+look-ahead+logic for clarification

Comment (szabolcs-keresztes) [reply src/functions/get/domain/config.mock.ts]: That's fine, however from the mobile app's point of view `numberOfDaysToShow` is 4 if current date is Friday.

But this is not important at the moment, this data is not going to be used anywhere for now.

Comment (buzzub) [reply src/functions/get/domain/config.mock.ts]: Not if it's Easter 😂 If today is Thursday and tomorrow is Good Friday and Monday is Easter Monday, the app should display Thursday and Tuesday - that's 6 days.

**Pull request merged**